Textfield/Textfield.jsx

```diff
@@ -1,50 +1,156 @@
+  }
+}
+
+/**
  * @prop fullwidth = false
  * @prop multiline = false
+ * @prop dense = false
+ * @prop disabled = false
  * @prop type = 'text'
  */
-export default class Textfield extends MaterialComponent {
+class TextfieldInput extends MaterialComponent {
   constructor() {
     super();
     this.componentName = "textfield";
+    this._mdcProps = ["fullwidth", "multiline", "dense", "disabled", "box"];
     this.state = {
       showFloatingLabel: false
     };
-    this._mdcProps = ["fullwidth", "multiline"];
   }
   componentDidMount() {
-    this.setState({
-      showFloatingLabel: true
-    });
-    this.MDComponent = new MDCTextfield(this.control);
+    this.setState(
+      {
+        showFloatingLabel: true
+      },
+      () => {
+        this.MDComponent = new MDCTextfield(this.control);
+      }
+    );
   }
   componentWillUnmount() {
     this.MDComponent.destroy && this.MDComponent.destroy();
   }
   materialDom(allprops) {
     const { className, ...props } = allprops;
-    let labelClass = ["mdc-textfield__label"];
-    if (props.value) {
-      labelClass.push("mdc-textfield__label--float-above");
-    }
+
     return (
       <div className={className} ref={control => (this.control = control)}>
         {props.multiline
-          ? <textarea className="mdc-textfield__input" {...props} />
+          ? <textarea
+              className="mdc-textfield__input"
+              aria-controls={props.id + "-helptext"}
+              {...props}
+            />
           : <input
               type={props.type || "text"}
               className="mdc-textfield__input"
+              aria-controls={props.id + "-helptext"}
               {...props}
             />}
         {props.label &&
           this.state.showFloatingLabel &&
-          <label className={labelClass.join(" ")}>
+          <Label for={props.id}>
             {props.label}
+          </Label>}
+      </div>
+    );
+  }
+}
+
+/**
+ * @prop fullwidth = false
+ * @prop multiline = false
+ * @prop dense = false
+ * @prop disabled = false
+ * @prop type = 'text'
+ * @prop helptext = ''
+ * @prop helptextPersistent = false
+ * @prop helptextValidationMsg = false
+ */
+export default class Textfield extends Component {
+  constructor() {
+    super();
+    this.id = Textfield.uid();
+    this.state = {
+      showFloatingLabel: false
+    };
+  }
+
+  componentDidMount() {
+    this.setState({
+      showFloatingLabel: true
+    });
+  }
+
+  static uid() {
+    if (!this.uidCounter) {
+      this.uidCounter = 0;
+    }
+    return ++this.uidCounter;
+  }
+
+  render(allprops, { showFloatingLabel }) {
+    const {
+      id,
+      className,
+      helptextPersistent,
+      helptextValidationMsg,
+      ...props
+    } = allprops;
+    const tfId = id || "tf-" + this.id;
+
+    // Helper text
+    const helptextProps = {
+      persistent: helptextPersistent,
+      "validation-msg": helptextValidationMsg
+    };
+
+    return (
+      <div className={className}>
```

Comment:
this will result into div->div->input which breaks us away from mdc-web
put the `{showFloatingLabel &&` check outside and put the extra div only incase of no js